### PR TITLE
Fixing on contact start and end being called every 20 frames. New behaviour for if the object is still colliding

### PR DESCRIPTION
**Pull request by BryceNewell-UL**
## Summary

Adds new logic for making sure that an object is still colliding with the bone which means the OnContactEnd and OnContactBegin aren't called every 20 frames. Will now be called when the object enters and exits collision, even when it is only the finger tip colliding.

## Contributor Tasks

_These tasks are for the merge request creator to tick off when creating a merge request._

- [ ] Pair review with a member of the QA team.
- [x] Add any release testing considerations to the MR for the next release.
- [x] Check any relevant CHANGELOG files have been updated.
- [x] Ensure documentation requirements are met e.g., public API is commented.
- [x] Consider any licensing/other legal implications for this MR e.g. notices required by any new libraries.
- [x] Add any relevant labels such as `breaking` to this MR.
- [ ] If this MR closes a Jira issue, make sure the fix version on the JIRA issue is set to the correct one.

## Reviewer Tasks

_Add any instructions or tasks for the reviewer such as specific test considerations before this can be merged._

[Use emojis in review threads to communicate intent and help contributors.](https://github.com/ultraleap/UnityPlugin/blob/develop/CONTRIBUTING.md#review-threads)

- [x] Code reviewed.
- [x] Non-code assets e.g. Unity assets/scenes reviewed.
- [ ] Documentation has been reviewed. Includes checking documentation requirements are met and not missing e.g., public API is commented.
- [ ] Checked and agree with release testing considerations added to MR for the next release.

## Closes JIRA Issue

_If this MR closes any JIRA issues list them below in the form `Closes PROJECT-#`_

### CODE REVIEW

File: Packages/Tracking/Interaction Engine/Runtime/Scripts/Internal/ContactBone.cs [hunk 1]
```diff
@@ -126,13 +129,37 @@ void OnDestroy()
...
+                    interactionController.NotifyContactBoneCollisionExit(this, contacting.Key);
+
+                    ContactingObjectsToRemove.Add(contacting.Key);
+                }
+            }
+
+            foreach(var removed in ContactingObjectsToRemove)
+            {
+                contactingInteractionBehaviours.Remove(removed);
+            }
```
Comment (MattGrayUL) [inline Packages/Tracking/Interaction Engine/Runtime/Scripts/Internal/ContactBone.cs:147]: Is populating a 'to remove' list necessary here? can we remove them in the first loop if it's a `for` loop that iterates backwards through the `contactingStayInteractionBehaviours` list instead?

**Comment by MattGrayUL** [comment]
This PR requires a changelog entry :)

File: Packages/Tracking/Interaction Engine/Runtime/Scripts/Internal/ContactBone.cs [hunk 2]
```diff
@@ -116,6 +117,9 @@ public float width
 
         Dictionary<IInteractionBehaviour, float> contactingInteractionBehaviours = new Dictionary<IInteractionBehaviour, float>();
 
+        List<IInteractionBehaviour> contactingStayInteractionBehaviours = new List<IInteractionBehaviour>();
+        List<IInteractionBehaviour> ContactingObjectsToRemove = new List<IInteractionBehaviour>();
```
Comment (MattGrayUL) [inline Packages/Tracking/Interaction Engine/Runtime/Scripts/Internal/ContactBone.cs:121]: this can be removed :)

Comment (BryceNewell-UL) [reply Packages/Tracking/Interaction Engine/Runtime/Scripts/Internal/ContactBone.cs]: Have removed it now

**Comment by BryceNewell-UL** [comment]
> This PR requires a changelog entry :)

PR now has a changelog entry ready for review

**Review by MattGrayUL — APPROVED** [review]

**Comment by MattGrayUL** [comment]
tested this with a log when oncontactend and oncontactbegin is called, this change stops it from repeatedly calling when one (or any number of) bone is contacting

Also tested the IE example scenes which all continue to work as expected, particularly focused on softly pushing objects to exacerbate any issues this may have cause, none were found

**Pull request merged**